src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -102,19 +110,26 @@ protected void postAnalyze(ControlFlowGraph cfg) {
                 if (node instanceof MethodInvocationNode) {
                   MethodInvocationNode methodInvocationNode = (MethodInvocationNode) node;
 
-                  System.out.printf("Method Invocation: %s\n", methodInvocationNode);
-                  System.out.printf("Operands: %s\n\n", methodInvocationNode.getOperands());
-
                   if (methodInvocationNode.getOperands().size() >= 2) {
                     MethodAccessNode methodAccessNode = methodInvocationNode.getTarget();
-                    Node fieldAccessNode = methodInvocationNode.getArgument(0);
+                    Node methodArgumentNode = methodInvocationNode.getArgument(0);
 
                     if (isBindMethod(methodInvocationNode.getTree())) {
                       // Class that is being bound - put in knownBindings
                       AnnotatedTypeMirror boundClassTypeMirror =
-                          this.getAnnotatedType(fieldAccessNode.getTree());
+                          this.getAnnotatedType(methodArgumentNode.getTree());
+
+                      List<String> classNames =
+                          AnnotationUtils.getElementValueArray(
+                              boundClassTypeMirror.getAnnotation(),
+                              classValValueElement,
+                              String.class);
```

Comment:
It's worth knowing that this call will fail if `getAnnotation()` returns any annotation other than `@ClassVal`, which is possible. We'll need to handle those cases eventually, so I suggest adding a TODO.